client/extensions/woocommerce/app/store-stats/utils.js

```diff
@@ -145,3 +173,59 @@ export function getDelta( deltas, selectedDate, stat ) {
 	const selectedDeltas = find( deltas, item => item.period === selectedDate );
 	return selectedDeltas[ stat ];
 }
+
+/**
+ * Given a date, an array of data, and a stat, return a delta object for the specific stat.
+ *
+ * @param {array} data - an array of API data, must contain at least 3 rows
+ * @param {string} selectedDate - string of date in 'YYYY-MM-DD'
+ * @param {string} stat - string of stat to be referenced
+ * @param {string} unit - unit/period format for the data provided
+ * @return {object} - Object containing data from calculateDelta
+ */
+export function getDeltaFromData( data, selectedDate, stat, unit ) {
+	if ( data.length < 3 ) {
```

Comment:
Yeah seems here we might be able to add a `! data && Array.isArray( data )` to simplify some of the usage of this in the components.